diff --git a/Engine/Source/Runtime/Renderer/Private/PostProcess/PostProcessing.cpp b/Engine/Source/Runtime/Renderer/Private/PostProcess/PostProcessing.cpp
index 105d8eb0eb7c..30635430f9a5 100644
--- a/Engine/Source/Runtime/Renderer/Private/PostProcess/PostProcessing.cpp
+++ b/Engine/Source/Runtime/Renderer/Private/PostProcess/PostProcessing.cpp
@@ -225,7 +225,7 @@ FDefaultTemporalUpscaler::FOutputs AddThirdPartyTemporalUpscalerPasses(
 	ThirdPartyInputs.SceneColor = Inputs.SceneColor;
 	ThirdPartyInputs.SceneDepth = Inputs.SceneDepth;
 	ThirdPartyInputs.SceneVelocity = Inputs.SceneVelocity;
-	ThirdPartyInputs.EyeAdaptationTexture = AddCopyEyeAdaptationDataToTexturePass(GraphBuilder, View);
+	ThirdPartyInputs.EyeAdaptationTexture = Inputs.SceneColor.Texture;// AddCopyEyeAdaptationDataToTexturePass(GraphBuilder, View);
 	
 	if (View.PrevViewInfo.ThirdPartyTemporalUpscalerHistory && View.PrevViewInfo.ThirdPartyTemporalUpscalerHistory->GetDebugName() == UpscalerName)
 	{
