diff --git a/Engine/Source/Runtime/Renderer/Private/VelocityRendering.cpp b/Engine/Source/Runtime/Renderer/Private/VelocityRendering.cpp
index ae0683b1fba1..01d07d4e74a9 100644
--- a/Engine/Source/Runtime/Renderer/Private/VelocityRendering.cpp
+++ b/Engine/Source/Runtime/Renderer/Private/VelocityRendering.cpp
@@ -55,6 +55,13 @@ static TAutoConsoleVariable<int32> CVarParallelVelocity(
 	ECVF_RenderThreadSafe
 	);
 
+static TAutoConsoleVariable<int32> CVarAlwaysRenderVelocity(
+	TEXT("r.AlwaysRenderVelocity"),
+	1,
+	TEXT("Force to use velocity\n"),
+	ECVF_ReadOnly | ECVF_RenderThreadSafe);
+
+
 DECLARE_GPU_DRAWCALL_STAT_NAMED(RenderVelocities, TEXT("Render Velocities"));
 
 /** Validate that deprecated CVars are no longer set. */
@@ -170,6 +177,11 @@ EMeshPass::Type GetMeshPassFromVelocityPass(EVelocityPass VelocityPass)
 
 bool FDeferredShadingSceneRenderer::ShouldRenderVelocities() const
 {
+	if (CVarAlwaysRenderVelocity.GetValueOnAnyThread() != 0)
+	{
+		return true;
+	}
+
 	if (!FVelocityRendering::IsVelocityPassSupported(ShaderPlatform) || ViewFamily.UseDebugViewPS())
 	{
 		return false;
@@ -214,6 +226,11 @@ bool FDeferredShadingSceneRenderer::ShouldRenderVelocities() const
 
 bool FMobileSceneRenderer::ShouldRenderVelocities() const
 {
+	if (CVarAlwaysRenderVelocity.GetValueOnAnyThread() != 0)
+	{
+		return true;
+	}
+
 	if (!FVelocityRendering::IsVelocityPassSupported(ShaderPlatform) || ViewFamily.UseDebugViewPS() || !PlatformSupportsVelocityRendering(ShaderPlatform))
 	{
 		return false;
diff --git a/Engine/Source/Runtime/SlateCore/Public/Rendering/SlateRenderer.h b/Engine/Source/Runtime/SlateCore/Public/Rendering/SlateRenderer.h
index 8bdcd5f732e3..37d9ae6ac5ea 100644
--- a/Engine/Source/Runtime/SlateCore/Public/Rendering/SlateRenderer.h
+++ b/Engine/Source/Runtime/SlateCore/Public/Rendering/SlateRenderer.h
@@ -11,6 +11,7 @@
 #include "Fonts/FontTypes.h"
 #include "Types/SlateVector2.h"
 #include "PixelFormat.h"
+#include <functional>
 
 class FRHITexture;
 class FRenderTarget;
@@ -284,6 +285,13 @@ public:
 	DECLARE_MULTICAST_DELEGATE_OneParam(FOnSlateWindowDestroyed, void*);
 	FOnSlateWindowDestroyed& OnSlateWindowDestroyed() { return OnSlateWindowDestroyedDelegate; }
 
+	using FDrawWindowsRenderThread = std::function<void()>;
+	DECLARE_MULTICAST_DELEGATE_OneParam(FOnDrawWindows_RenderThread, FDrawWindowsRenderThread);
+	FOnDrawWindows_RenderThread& OnDrawWindows_RenderThread() { return DrawWindowsDelegate; }
+
+	DECLARE_TS_MULTICAST_DELEGATE_ThreeParams(FOnDrawWindow_RenderThread, FRDGBuilder&, FRHICommandListImmediate&, FTextureRHIRef&);
+	FOnDrawWindow_RenderThread& OnDrawWindow_RenderThread() { return DrawWindowDelegate; }
+
 	/**
 	 * Called on the game thread right before a window backbuffer is about to be resized
 	 * @param Pointer to the API specific backbuffer type
@@ -589,6 +597,9 @@ protected:
 	FOnPreResizeWindowBackbuffer PreResizeBackBufferDelegate;
 	FOnPostResizeWindowBackbuffer PostResizeBackBufferDelegate;
 
+	FOnDrawWindows_RenderThread DrawWindowsDelegate;
+	FOnDrawWindow_RenderThread DrawWindowDelegate;
+
 	FOnBackBufferReadyToPresent OnBackBufferReadyToPresentDelegate;
 	FOnAddBackBufferReadyToPresentPass OnAddBackBufferReadyToPresentPassDelegate;
 
diff --git a/Engine/Source/Runtime/SlateRHIRenderer/Private/SlateRHIRenderer.cpp b/Engine/Source/Runtime/SlateRHIRenderer/Private/SlateRHIRenderer.cpp
index 851a5ffd5520..ee9c3bb403d0 100644
--- a/Engine/Source/Runtime/SlateRHIRenderer/Private/SlateRHIRenderer.cpp
+++ b/Engine/Source/Runtime/SlateRHIRenderer/Private/SlateRHIRenderer.cpp
@@ -1144,6 +1144,10 @@ void FSlateRHIRenderer::DrawWindows_RenderThread(FRHICommandListImmediate& RHICm
 			for (int32 NumWindows = 0; NumWindows < 8 && WindowIndex < Windows.Num(); ++NumWindows, ++WindowIndex)
 			{
 				const FSlateDrawWindowPassInputs& DrawWindowPassInputs = Windows[WindowIndex];
+				FTextureRHIRef SlateRenderTarget = RHIGetViewportBackBuffer(DrawWindowPassInputs.ViewportInfo->ViewportRHI);
+				
+				DrawWindowDelegate.Broadcast(GraphBuilder, RHICmdList, SlateRenderTarget);
+
 				WindowPresentCommands.Emplace(DrawWindowPassInputs, DrawWindow_RenderThread(GraphBuilder, DrawWindowPassInputs));
 			}
 
@@ -1427,9 +1431,16 @@ void FSlateRHIRenderer::DrawWindows_Private(FSlateDrawBuffer& WindowDrawBuffer)
 
 	if (!DrawWindowsCommand->IsEmpty())
 	{
+
 		ENQUEUE_RENDER_COMMAND(SlateDrawWindowsCommand)([this, DrawWindowsCommand = MoveTemp(DrawWindowsCommand)](FRHICommandListImmediate& RHICmdList)
 		{
-			DrawWindows_RenderThread(RHICmdList, DrawWindowsCommand->Windows, DrawWindowsCommand->DeferredUpdates);
+				auto DrawWindows_RenderThreadFx = [&]()
+					{
+						DrawWindows_RenderThread(RHICmdList, DrawWindowsCommand->Windows, DrawWindowsCommand->DeferredUpdates);
+					};
+
+			DrawWindowsDelegate.Broadcast(DrawWindows_RenderThreadFx);
+			DrawWindows_RenderThreadFx();
 		});
 
 		check(DeferredUpdateContexts.IsEmpty());
